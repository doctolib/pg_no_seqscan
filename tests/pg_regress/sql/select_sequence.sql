-- Test that indexed queries don't trigger errors

SET pg_no_seqscan.level = ERROR;

-- Querying a sequence should not error
CREATE SEQUENCE test_seq;
EXPLAIN SELECT last_value FROM test_seq;
SELECT last_value FROM test_seq;

-- cleanup
DROP SEQUENCE test_seq;
