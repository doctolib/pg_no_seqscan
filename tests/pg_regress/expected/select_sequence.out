-- Test that indexed queries don't trigger errors
SET pg_no_seqscan.level = ERROR;
-- Querying a sequence should not error
CREATE SEQUENCE test_seq;
EXPLAIN SELECT last_value FROM test_seq;
                       QUERY PLAN                       
--------------------------------------------------------
 Seq Scan on test_seq  (cost=0.00..1.01 rows=1 width=8)
(1 row)

SELECT last_value FROM test_seq;
 last_value 
------------
          1
(1 row)

-- cleanup
DROP SEQUENCE test_seq;
